balancer/pickfirst/pickfirstleaf/pickfirstleaf.go

```diff
@@ -600,11 +639,11 @@ func (b *pickfirstBalancer) updateSubConnState(sd *scData, newState balancer.Sub
 	// a transport is successfully created, but the connection fails
 	// before the SubConn can send the notification for READY. We treat
 	// this as a successful connection and transition to IDLE.
-	if (b.state == connectivity.Ready && newState.ConnectivityState != connectivity.Ready) || (oldState == connectivity.Connecting && newState.ConnectivityState == connectivity.Idle) {
+	if (oldState == connectivity.Ready && newState.ConnectivityState != connectivity.Ready) || (oldState == connectivity.Connecting && newState.ConnectivityState == connectivity.Idle) {
```

Comment:
If `oldState` is `Ready`, doesn't `newState` _have_ to be `Idle`?

And if `newState` is `Idle`, the previous state _must have_ been `Ready`, right?  We may have not gotten the update (because of the Connecting->Ready->Idle race), but....can this whole thing be replaced by just `if newState.ConnectivityState == connectivity.Idle`?

Reply:
As discussed offline, please change to look like we want it to look (if we didn't have the bug) and add a clause that covers the bug:

```go
if oldState == Ready || (oldState == Connecting && newState == Idle) {
```

Reply:
(and add a TODO to remove the second half when the bug is fixed please)

Reply:
Done.